Unit-Tests/Replication_Tests.m

```diff
@@ -969,4 +970,96 @@ - (void) test16_Restart {
     AssertEq(puller.status, kCBLReplicationStopped);
 }
 
+- (void)test17_RemovedRevision {
+    NSURL* remoteDbURL = [self remoteTestDBURL: kRemovedRevDBName];
+    if (!remoteDbURL)
+        return;
+    [self eraseRemoteDB: remoteDbURL];
+
+    NSError* error;
+
+    // Create a new document with grant = true:
```

Comment:
This seems too complex for a unit test. It would be fine to just manually create a revision with a `"_removed":true` property and then verify that it doesn't get pushed.